pkg/netceptor/internal/shared.go

```diff
@@ -0,0 +1,68 @@
+package netceptor_internal
```

Comment:
```suggestion
package netceptor
```
You don't need to add the `_internal` to mark these functions as an internal to the netceptor package. Simply putting it in the internal folder is enough. 😃